lib/connect/ws.js

```diff
@@ -32,15 +40,14 @@ function buildBuilder (client, opts) {
     wsOpt.protocol = 'mqttv3.1'
   }
 
-  if (opts.protocol === 'wss') {
-    wssProperties.forEach(function (prop) {
-      if (opts.hasOwnProperty(prop)) {
-        wsOpt[prop] = opts[prop]
-      }
-    })
-  }
-
-  return websocket(url, wsOpt)
+  /*delete some mqtt overhead*/
+  mqttProperties.forEach(function (prop) {
+    if (opts.hasOwnProperty(prop)) {
+      delete opts[prop];
```

Comment:
I'm not sure I understand the reason for this. Is this because you want to prevent some options to be set? (i.e. the `mqttProperties` list)